source/declarations.tex

```diff
@@ -1896,29 +1896,35 @@
 
 \pnum
 A type \tcode{T} containing a placeholder type,
-and a corresponding initializer $E$,
+and a corresponding \grammarterm{initializer-clause} $E$,
 are determined as follows:
 \begin{itemize}
 \item
-for a non-discarded \tcode{return} statement that occurs
+For a non-discarded \tcode{return} statement that occurs
 in a function declared with a return type
 that contains a placeholder type,
 \tcode{T} is the declared return type
-and $E$ is the operand of the \tcode{return} statement.
+and $E$ is the operand of the \tcode{return} statement;
+if the operand is an \grammarterm{expression} $X$
+that is not an \grammarterm{assignment-expression}\iref{expr.comma},
+$E$ is \tcode{($X$)}.
```

Comment:
Ugh, this is awkward. This does the wrong thing for:
```
decltype(auto) f(int n) { return n; }
```
... where we should deduce `int`, but if `E` is `(n)` then we'll deduce `int&` below.

Reply:
"n" is (indirectly) an assignment-expression.

Reply:
@zygoloid: does that address your concerns? Is this still too subtle?